Tools/CISupport/ews-build/steps.py

```diff
@@ -5055,15 +5079,32 @@ def run(self, BufferLogObserverClass=logobserver.BufferLogObserver):
             self.summary = 'Patches have no commit message'
             return rc
 
-        log_text = self.log_observer.getStdout().rstrip()
+        self.contributors, errors = Contributors.load(use_network=True)
+        for error in errors:
+            print(error)
+            self._addToLog('stdio', error)
+
+        reviewers, log_text = self.extract_reviewers(self.log_observer.getStdout())
+        log_text = log_text.rstrip()
         if any(['ChangeLog' in file for file in self._files()]):
             self.summary = 'ChangeLog modified, WebKit only allows commit messages'
             rc = FAILURE
         elif log_text:
             self.summary = log_text
             rc = FAILURE
         elif rc == SUCCESS:
-            self.summary = 'Validated commit message'
+            if reviewers and not self.contributors:
+                self.summary = "Failed to load contributors.json, can't validate reviewers"
+                rc = FAILURE
+            elif reviewers and not all([self.is_reviewer(reviewer) for reviewer in reviewers])
+                self.summary = '{} is not a reviewer'
+                for reviewer in reviewers:
+                    if not self.is_reviewer(reviewer):
+                        self.summary = self.summary.format(reviewer)
+                        break
+                rc = FAILURE
+            else:
```

Comment:
what if reviewers is None?

Reply:
That means you have some variant of "Unreviewed", or you wouldn't have passed the 2nd check in this step.

Reply:
ok